build.c

```diff
@@ -115,13 +115,43 @@ queue(struct edge *e)
 	*front = e;
 }
 
+static void
+computedirty(struct edge *e, struct node *newest)
+{
+	struct node *n;
+	size_t i;
+	bool generator, restat;
+
+	/* all outputs are dirty if any are older than the newest input */
+	generator = edgevarbool(e, "generator");
+	restat = edgevarbool(e, "restat");
```

Comment:
Just checking, does ninja treat set-but-empty `generator` and `restat` as false?